app/models/metric.rb

```diff
@@ -239,7 +245,12 @@ def associate_to_service_of_parent
 
   def fill_owner
     return true if owner_type?
-    self.owner_id = service_id
-    self.owner_type = 'Service'
+
+    if new_record? && child?
+      self.owner = parent.owner
+    else
+      self.owner_id = service_id
+      self.owner_type = 'Service'
```

Comment:
So if it is an old record OR it is hits, the owner is always a service and not a backend api. Is this still correct ? (I genuinely ask because I really don't know, but it does not sound correct to me)

Reply:
Ok, @hallelujah told me that it is correct 👍